File: org.kevoree.modeling.microframework/src/test/java/org/kevoree/modeling/microframework/test/LookupTest.java
package org.kevoree.modeling.microframework.test;

import org.junit.Assert;
import org.junit.Test;
import org.kevoree.modeling.api.Callback;
import org.kevoree.modeling.api.KObject;
import org.kevoree.modeling.microframework.test.cloud.CloudUniverse;
import org.kevoree.modeling.microframework.test.cloud.CloudModel;
import org.kevoree.modeling.microframework.test.cloud.CloudView;
import org.kevoree.modeling.microframework.test.cloud.Node;

/**
 * Created by duke on 10/23/14.
 */
public class LookupTest {

    @Test
    public void lookupTest() throws Exception {

        // MemoryKContentDeliveryDriver.DEBUG = true;
        //  MultiLayeredMemoryCache.DEBUG = true;

        final CloudModel cloudModel = new CloudModel();
        cloudModel.connect(null);
        final CloudUniverse dimension0 = cloudModel.newUniverse();
        CloudView t0 = dimension0.time(0l);
        final Node node = t0.createNode();
        node.setName("n0");
        t0.setRoot(node,null);
        //Assert.assertTrue(node.isRoot());
        cloudModel.manager().getRoot(t0.universe(),t0.now(), new Callback<KObject>() {
            @Override
            public void on(KObject resolvedRoot) {
                Assert.assertEquals(node, resolvedRoot);
            }
        });
        //Assert.assertTrue(node.isRoot());

        cloudModel.save(new Callback<Throwable>() {
            @Override
            public void on(Throwable error) {
                cloudModel.discard(new Callback<Throwable>() {
                    @Override
                    public void on(Throwable throwable) {

                        final CloudModel universe2 = new CloudModel();
                        universe2.setContentDeliveryDriver(cloudModel.manager().cdn());
                        CloudUniverse dimension0_2 = universe2.universe(dimension0.key());
                        final CloudView t0_2 = dimension0_2.time(0l);

                        universe2.manager().getRoot(t0_2.universe(),t0_2.now(), new Callback<KObject>() {
                            @Override
                            public void on(KObject resolvedRoot) {
                                Assert.assertEquals(node.uuid(), resolvedRoot.uuid());
                            }
                        });
                        t0_2.lookup(node.uuid(),new Callback<KObject>() {
                            @Override
                            public void on(final KObject resolved) {
                                Assert.assertNotNull(resolved);
                                t0_2.lookup(node.uuid(),new Callback<KObject>() {
                                    @Override
                                    public void on(KObject resolved2) {
                                        Assert.assertEquals(resolved, resolved2);
                                    }
                                });
                                //Assert.assertTrue(resolved.isRoot());
                            }
                        });
                    }
                });


            }
        });

    }

}

File: org.kevoree.modeling.microframework.typescript/src/test/ts/org/kevoree/modeling/microframework/test/LookupTest.ts
///<reference path="../../../../junit/Assert.ts"/>
///<reference path="../../../../junit/Test.ts"/>
///<reference path="../../api/Callback.ts"/>
///<reference path="../../api/KObject.ts"/>
///<reference path="../../api/data/MemoryKDataBase.ts"/>
///<reference path="cloud/CloudDimension.ts"/>
///<reference path="cloud/CloudUniverse.ts"/>
///<reference path="cloud/CloudView.ts"/>
///<reference path="cloud/Node.ts"/>

class LookupTest {

  public lookupTest(): void {
    var dataBase: MemoryKDataBase = new MemoryKDataBase();
    var universe: CloudUniverse = new CloudUniverse(dataBase);
    universe.newDimension({on:function(dimension0: CloudDimension){
    var t0: CloudView = dimension0.time(0);
    var node: Node = t0.createNode();
    node.setName("n0");
    t0.setRoot(node);
    Assert.assertTrue(node.isRoot());
    universe.storage().getRoot(t0, {on:function(resolvedRoot: KObject){
    Assert.assertEquals(node, resolvedRoot);
}});
    Assert.assertTrue(node.isRoot());
    dimension0.save({on:function(e: Throwable){
    var universe2: CloudUniverse = new CloudUniverse(dataBase);
    universe2.dimension(dimension0.key(), {on:function(dimension0_2: CloudDimension){
    var t0_2: CloudView = dimension0_2.time(0);
    t0_2.lookup(node.uuid(), {on:function(resolved: KObject){
    t0_2.lookup(node.uuid(), {on:function(resolved2: KObject){
    Assert.assertEquals(resolved, resolved2);
}});
    universe2.storage().getRoot(t0_2, {on:function(resolvedRoot: KObject){
    Assert.assertEquals(resolved, resolvedRoot);
}});
    Assert.assertTrue(resolved.isRoot());
}});
}});
}});
}});
  }

}

